_plugins/sidebar.rb

```diff
@@ -0,0 +1,216 @@
+        sidebar = render(sanitized_filename)
+        file = File.open(filename) { |f| Nokogiri::HTML(f) }
+        file.xpath('//*[@id="dx-sidebar-main-nav-ul"]').each do |location|
+          location.inner_html = sidebar
+        end
+        File.open(filename, 'w') { |f| f.write(file.to_html(encoding: 'UTF-8')) }
+      end
+
+      # File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
+    end
+
+    private
+
+    def sanitize_filename(filename)
+      sanitized_filename = filename.match(/(?m)(?<=\b_site).*$/)[0]
+      sanitized_filename = sanitized_filename.gsub('index.html', '')
+      sanitized_filename.gsub('.html', '')
+    end
+
+    def generateSubgroup(filename, key, value, all_subgroups, level)
+      title = value[:title].split('–').last
+      subsubgroup_list = all_subgroups.select do |subsubgroup_key, _subsubgroup_value|
+        subsubgroup_key.include? key and subsubgroup_key != key and \
+          key.split('/').length > level
+      end
+
+      subgroup = ''
+      has_subgroups = !all_subgroups.empty?
+      if value[:headers].any? || !subsubgroup_list.empty?
+        if has_subgroups
+          url = value[:url]
+          active = active?(filename, url, true)
+          # puts "#{url}, #{filename}, #{key}" if active
+          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
+          subgroup << "<li class=\"#{item_class}\">"
+          subgroup << "<div class=\"nav-subgroup-heading\"><i class=\"material-icons\">arrow_right</i><a href=\"#{url}\">#{title}</a></div>"
+          subgroup << '<ul class="nav-ul">'
+
+          if subsubgroup_list.empty?
+            value[:headers].each do |header|
+              subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
+            end
+          else
+            subgroup_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
+            subgroup << "<li class=\"#{subgroup_leaf_class}\"><a href=\"#{value[:url]}\">#{title} overview</a></li>"
+
+            subsubgroup_list.each do |subsubgroup_key, subsubgroup_value|
+              subgroup << generateSubgroup(filename, subsubgroup_key, subsubgroup_value, subsubgroup_list, 3)
+            end
+          end
+
+          subgroup << '</ul>'
+          subgroup << '</li>'
+        else
+          value[:headers].each do |header|
+            subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
+          end
+        end
+      else
+        subgroup << if has_subgroups
+                      "<li class=\"nav-leaf nav-subgroup-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
+                    else
+                      "<li class=\"nav-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
+                    end
+      end
+
+      subgroup
+    end
+
+    def render(filename)
```

Comment:
Codacy found an issue: [Cyclomatic complexity for render is too high. [7/6]](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=6023491)